COMP 120 Tinkering Graphics-JH182233/Program/Option.py

```diff
@@ -0,0 +1,205 @@
+import time
+
+
+
+#This is the stored pictures, calling "global" so the entire program can use them.
+setMediaPath("F:\COMP 120 Tinkering Graphics-JH182233\Pictures")
+global picture1 
+picture1 = makePicture("Tree-Four-Seasons.jpg")
+global picture2 
+picture2 = makePicture("Tree-Winter.jpg")
+global picture3 
+picture3 = makePicture("Tree-Spring.jpg")
+global picture4 
+picture4 = makePicture("Tree-Summer.jpg")
+global picture5 
+picture5 = makePicture("Tree-Autumn.jpg")  
+global picture6
+picture6 = makePicture("tree.jpg")
+global picture7
+picture7 = makePicture("test.jpg")
+global picM
+picM = makeEmptyPicture(513, 556, black)
+
+
+
+#The main function for the program
+def main():
+#create an empty picture and populate it with others 
+    startX = 0
+    startY = 0
+    global pic
+    pic = makeEmptyPicture(1025, 1110, black) 
+    repaint (pic)
+    time.sleep(1)  
+    copyInto(picture2, pic, startX, startY)
+    repaint (pic)
+    
+    startX = 512
+    startY = 0    
+    time.sleep(1) 
+    copyInto(picture3, pic, startX, startY)
+    repaint (pic)
+    
+    startX = 0
+    startY = 554  
+    time.sleep(1)  
+    copyInto(picture4, pic, startX, startY)
+    repaint (pic)
+    
+    startX = 512
+    startY = 554  
+    time.sleep(1)  
+    copyInto(picture5, pic, startX, startY)
+    repaint (pic)
+    request()
+    
+    
+    
+#Now the program will ask for a season that you would like to view
+def request():
+    startX = 230
+    startY = 300   
+    season = requestString("What season would you like to activate?")
+    if String(season) =="Winter":
+     blackoutleft()
+     copyInto(picture2, pic, startX, startY)
+     repaint(pic)
+     time.sleep(2)
+     request2()
+    elif String(season) =="Spring":
+     blackoutleft()  
+     copyInto(picture3, pic, startX, startY)
+     repaint(pic)
+     time.sleep(2)
+     request2()
+    elif String(season) =="Summer":
+     blackoutleft()   
+     copyInto(picture4, pic, startX, startY)
+     repaint(pic)
+     time.sleep(2)
+     request2()
+    elif String(season) =="Autumn":
+     blackoutleft()   
+     copyInto(picture5, pic, startX, startY)
+     repaint(pic)
+     time.sleep(2)
+     request2()
+    else:
+     blackout()
+     fourseasons()
+     time.sleep(2)
+     
+    
+
+#another request, a simple Y/N question to see if you want to continue, it will either end for "N" or continue if "Y"    
+def request2():
+    endSeason = requestString("Would you like to view more Y/N?")
+    if String(endSeason) == "Y":
+     copyInto(picture1, pic, 0, 0)
+     repaint(pic)
+     playseason()
+    elif String(endSeason) == "N":
+     blackouttop()
+     copyInto(picture1, pic, 0, 0)
+     repaint(pic)
+     showInformation("Thank you for using BioSeasons")
+     blackouttop()
+    else:       
+     request()
+     
+     
+
+#Thie function will play through the seasons while calling on the merge function to produce a better flow.
+def playseason():
```

Comment:
It may make the code more readable and adaptable to separate this into separate functions for each season, and then call each of them in turn. Currently, its impossible to tell which picture is which season without going to the top of the code and viewing the files. If you had a "showWinter" function, for example, it would be clear which part of the code needs modifying if you wanted to change something about winter. Separating the functions also makes it easy to change the order.